File: 2839.py
# n = int(input())

# dp = [int(1e9)]*(n+1)


# for i in range(3,len(dp)):
# 	dp[i] = min(dp[i-1]+i,i//3,i//5)
# print(dp)
# print(dp[n]+1)


sugar = int(input())

bag = 0
while sugar >= 0 :
	if sugar % 5 == 0 :  # 5의 배수이면
		bag += (sugar // 5)  # 5로 나눈 몫을 구해야 정수가 됨
		print(bag)
		break
	sugar -= 3
	bag += 1  # 5의 배수가 될 때까지 설탕-3, 봉지+1
else :
	print(-1)

File: 2839.js
// 3,5kg
input = require("fs").readFileSync("/dev/stdin");
bag = 0;

while (true) {
  if (input % 5 === 0) {
    console.log(bag + input / 5);
    break;
  }
  if (input < 0) {
    console.log(-1);
    break;
  }
  bag++;
  input -= 3;
}

// let input = require('fs').readFileSync('/dev/stdin');

// let count = 0;

// while (true) {
//   if (input % 5 === 0) {
//     console.log(input / 5 + count);
//     break;
//   }

//   if (0 > input) {
//     console.log(-1);
//     break;
//   }

//   count++;
//   input -= 3;
// }
